lib/sinon/stub.js

```diff
@@ -19,12 +19,15 @@ var functionToString = require("./util/core/function-to-string");
 var valueToString = require("./util/core/value-to-string");
 var wrapMethod = require("./util/core/wrap-method");
 
-function stub(object, property, funcOrDescriptor) {
-    if (!!funcOrDescriptor && typeof funcOrDescriptor !== "function" && typeof funcOrDescriptor !== "object") {
-        throw new TypeError("Custom stub should be a function or a property descriptor");
```

Comment:
This should probably not throw an error at this point in time, and rather print a deprecation warning using `lib/sinon/util/core/deprecated`. We can do the more drastic bit later on.